src/CommandBasedBackendRegisterInfo.cc

```diff
@@ -13,18 +106,180 @@ namespace ChimeraTK {
+    if(!j.contains(to_str(MAP_FILE_FORMAT_VERSION))) {
+      throw ChimeraTK::logic_error("Missing mapFileFormatVersion key in metadata");
+    }
+
+    throwIfHasInvalidJsonKey(j, metadataKeyStrs, "Map file metadata has unknown key");
+
+    if(int mapFileFormatVersion = j.value(to_str(MAP_FILE_FORMAT_VERSION), 0);
+        mapFileFormatVersion != requiredMapFileFormatVersion) {
+      throw ChimeraTK::logic_error("Incorrect map file format version " + std::to_string(mapFileFormatVersion) +
+          ", version " + std::to_string(requiredMapFileFormatVersion) + " required.");
+    }
+
+    m.defaultRecoveryRegister = RegisterPath(j.value(to_str(DEFAULT_RECOVERY_REGISTER), ""));
+
+    j.at(to_str(DELIM)).get_to(m.serialDelimiter);
+  }
+
+  /******************************************************************************************************************/
+
+  CommandBasedBackendRegisterInfo getJsonToCommandBasedBackendRegisterInfo(const json& j, const std::string& regKey) {
+    // regKey is the key (register path) whose value is j. This is needed information for debugging
+
+    if(not(j.contains(to_str(READ_CMD)) or j.contains(to_str(WRITE_CMD)))) {
+      throw ChimeraTK::logic_error("A non-empty " + to_str(READ_CMD) + " or " + to_str(WRITE_CMD) +
+          " tags is required, and neither are present for register " + regKey);
+    }
+    else if((not j.contains(to_str(READ_CMD))) and j.at(to_str(WRITE_CMD)) == "") {
+      throw ChimeraTK::logic_error("A non-empty " + to_str(READ_CMD) + " or " + to_str(WRITE_CMD) +
+          " tags is required. writeCmd is blankd and readCmd is missing for register " + regKey);
     }
-    if(internalType == InternalType::UINT64) {
-      dataDescriptor = DataDescriptor(DataType::uint64);
+    else if((not j.contains(to_str(WRITE_CMD))) and j.at(to_str(READ_CMD)) == "") {
+      throw ChimeraTK::logic_error("A non-empty " + to_str(READ_CMD) + " or " + to_str(WRITE_CMD) +
+          " tags is required. readCmd is blankd and writeCmd is missing for register " + regKey);
     }
-    if(internalType == InternalType::DOUBLE) {
-      dataDescriptor = DataDescriptor(DataType::float64);
+
+    // Throw if there are responces without corresponding commands.
+    /*/
+    if(j.contains(to_str(READ_RESP)) and j.at(to_str(READ_RESP)) != "" and ( (not j.contains(to_str(READ_CMD))) or
+    j.at(to_str(READ_CMD) == ""))) { throw ChimeraTK::logic_error("A non-empty "+to_str(READ_RESP)+" without a non-empty
+    "+to_str(READ_CMD)+" for register "+regKey);
     }
-    if(internalType == InternalType::STRING) {
-      dataDescriptor = DataDescriptor(DataType::string);
+    if(j.contains(to_str(WRITE_RESP)) and j.at(to_str(WRITE_RESP)) != "" and ( (not j.contains(to_str(WRITE_CMD))) or
+    j.at(to_str(WRITE_CMD) == ""))) { throw ChimeraTK::logic_error("A non-empty "+to_str(WRITE_RESP)+" without a
+    non-empty "+to_str(WRITE_CMD)+" for register "+regKey);
+    }*/
```

Comment:
I did not fully read to this, but when we are done there should be no commented code section.